src/app/pages/your-virtrolio/your-virtrolio.component.ts

```diff
@@ -73,6 +93,39 @@ export class YourVirtrolioComponent implements OnInit {
     }
   }
 
+  shareLink() {
+    if (this.canShare()) {
+      this.navigator.share({
+        title: this.displayName + '\'s virtrolio!',
+        text: 'Visit this link to sign their virtrolio and send them a custom message!',
+        url: this.link,
+      })
+        .then(() => console.log('Successful share'))
+        .catch((error) => console.log('Error sharing', error));
+    } else {
+      alert('Share not supported on this device. Use a mobile device!');
+    }
+  }
+
+  /**
+   * Share the user's sharing link on the appropriate platform
+   * @param platform - platform on which to post the sharing link
+   */
+  postOnSocial(platform: string) {
+    const urlFriendlyLink = 'https%3A//virtrolio.web.app/signing?uid=' + this.authService.uid() + '%26key=' +
+      this.link.match(/key=([^&]*)/)[1];
+
+    if (platform === 'facebook') {
+      window.open('https://www.facebook.com/sharer/sharer.php?u=' + urlFriendlyLink, '_blank');
+    } else if (platform === 'twitter') {
+      window.open('https://twitter.com/intent/tweet?url=' + urlFriendlyLink + '&text=Send%20me%20a%20message!', '_blank');
+    } else if (platform === 'gmail') {
+      window.open('https://mail.google.com/mail/u/0/?view=cm&fs=1&tf=1&source=mailto&body=' + urlFriendlyLink, '_blank');
```

Comment:
You should add titles and bodies for these

Reply:
Twitter and Facebook don't have 'titles', but I'll put 'Send me a message!' into their bodies. I'll add a subject line for the email.

Reply:
https://stackoverflow.com/questions/5023602/facebook-share-link-can-you-customize-the-message-body-text

So facebook body text modification is a nono. Got Gmail to work though. That should be enough.